lib/Dialect/TTNN/Analysis/ShardSolver.cpp

```diff
@@ -561,9 +566,102 @@ bool ShardSolver::checkShardCompatible(
   return true;
 }
 
+// Preprocess ShardSolver search space to make a helper structure which links op
+// layout choices to global max core usage.
+// Example:
+// Lets assume simple case where layouts at same index are compatible for input
+// graph provided below.
+//
+//    Op0 ----- (4, 8, 2)
+//     |
+//    Op1 ----- (8, 4, 2)
+//    / \
+//   /   \
+//  Op2  Op3 -- (4, 4, 2) (4, 4, 2)
+//   \   /
+//    \ /
+//    Op4 ----- (2, 1, 1)
+//     |
+//    Op5 ----- (2, 1, 1)
+//
+// Here is how structure looks after preprocessing is complete:
+//
+//    Op0 ----- (24, 22, 10)
+//     |
+//    Op1 ----- (20, 14, 8)
+//    / \
+//   /   \
+//  Op2  Op3 -- (6, 5, 3) (6, 5, 3)
+//   \   /
+//    \ /
+//    Op4 ----- (4, 2, 2)
+//     |
+//    Op5 ----- (2, 1, 1)
+//
+// Global max of 24 core usage is achieved by selecting layout[0] for each Op.
+//
+// Returns map of op to vector of max core usage for each layout.
+llvm::DenseMap<Operation *, SmallVector<float, 64>>
+ShardSolver::produceMaxCoreUsage() {
```

Comment:
@svuckovicTT @nsmithtt  I think you guys wanted to do something similar in the Buda stack, so here it is. :)

Reply:
What does the tuple of numbers mean? `Op0 ----- (4, 8, 2)`

Reply:
Op0 legal layouts 0,1,2 ->(use) (4, 8, 2) cores